### DISCUSSION TITLE
WiFi hotspot homespan doesn't appear

### DISCUSSION THREAD

**Discussion by Adajaken** [HomeSpan/HomeSpan#84] [Q&A]
All functions are verified well except that WiFi hotspot “homespan” doesn’t show up. Changing LED from flickering 1 times per second to 2 times per seconds, or 3 times, 4 times via button are all tested, but WiFi hotspot never appear.  Why is that and how to solve?

**Comment by HomeSpan** [comment]
Can you check the following:  From within the Arduino IDE Serial Monitor, type 'A <return>' when connected to HomeSpan.  This should start the hot-spot and you should see the network HomeSpan-Setup show up under Settings->WiFi on your iPhone.  If this does not work, can you verify that the ESP32 can connect to you WiFi network manually by using the 'W' command?

Also, can you send me the log file that appears in the Arduino IDE Serial Monitor when you first start the ESP32 (I assume you are using an ESP32, and not one of the newer ESP32S2 or ESP32C3 chips?

**Reply by Adajaken** [reply]
After type "A" in Serial Monitor, I can find Wi-Fi hotspots nearby, and also find hotspot "HomeSpan" in iPhone, and everything works well after I connect to it.  However, I can still not call for WiFi hotspot "HomeSpan" by button. This is my problem, need to use button to find hotspot. How to solve?

Also, I confirm what I use is ESP32, not ESP32S2 or ESP32C3.

Below is  the log file that appears in the Arduino IDE Serial Monitor:

E (111) psram: PSRAM ID read error: 0xffffffff

************************************************************
Welcome to HomeSpan!
Apple HomeKit for the Espressif ESP-32 WROOM and Arduino IDE
************************************************************

** Please ensure serial monitor is set to transmit <newlines>

Message Logs:     Level 0
Status LED:       Pin 13
Device Control:   Pin 21
Sketch Version:   n/a
HomeSpan Version: 1.2.1
ESP-IDF Version:  v3.3.5-1-g85c43024c
ESP32 Board:      esp32
Sketch Compiled:  Jun  1 2021 05:38:29

Device Name:      HomeSpan LightBulb


*** HomeSpan Info ***

➤ Accessory:  AID=1
   ➟ Service AccessoryInformation:  IID=1, UUID=0x3E
      ⇨ Characteristic Name("My Table Lamp"):  IID=2, UUID=0x23
      ⇨ Characteristic Manufacturer("HomeSpan"):  IID=3, UUID=0x20
      ⇨ Characteristic SerialNumber("123-ABC"):  IID=4, UUID=0x30
      ⇨ Characteristic Model("120-Volt Lamp"):  IID=5, UUID=0x21
      ⇨ Characteristic FirmwareRevision("0.9"):  IID=6, UUID=0x52
      ⇨ Characteristic Identify(0):  IID=7, UUID=0x14
   ➟ Service HAPProtocolInformation:  IID=8, UUID=0xA2
      ⇨ Characteristic Version("1.1.0"):  IID=9, UUID=0x37
   ➟ Service LightBulb:  IID=10, UUID=0x43
      ⇨ Characteristic On(0):  IID=11, UUID=0x25

Configured as Bridge: NO

Service                         UUID         AID  IID  Update  Loop  Button  Linked Services
------------------------------  ----  ----------  ---  ------  ----  ------  ---------------
AccessoryInformation              3E           1    1      NO    NO      NO  -
HAPProtocolInformation            A2           1    8      NO    NO      NO  -
LightBulb                         43           1   10      NO    NO      NO  -

*** End Info ***

Generating SRP verification data for default Setup Code: 466-37-726
Setup Payload for Optional QR Code: X-HM://0052SD8YM    

Generating new random Accessory ID and Long-Term Ed25519 Signature Keys...
Initializing storage for Paired Controllers data...

Accessory ID:      EE:A7:C1:C4:4D:9D                               LTPK: F9F4D18516862B38BA0CE01EB748FABB21C604C3DC4E2B82755749F8A8EA8574
No Paired Controllers
Resetting Accessory Configuration number...

Accessory configuration has changed.  Updating configuration number to 1

*** WIFI CREDENTIALS DATA NOT FOUND.  YOU MAY CONFIGURE BY TYPING 'W <RETURN>'.

HomeSpan LightBulb is READY!

**Comment by Adajaken** [comment]
Dear,

After type "A" in Serial Monitor, I can find Wi-Fi hotspots nearby, and also find hotspot "HomeSpan" in iPhone, and everything works well after I connect to it.  However, I can still not call for WiFi hotspot "HomeSpan" by button. This is my problem, need to use button to find hotspot. How to solve?

Also, I confirm what I use is ESP32, not ESP32S2 or ESP32C3.

Below is  the log file that appears in the Arduino IDE Serial Monitor:

E (111) psram: PSRAM ID read error: 0xffffffff

************************************************************
Welcome to HomeSpan!
Apple HomeKit for the Espressif ESP-32 WROOM and Arduino IDE
************************************************************

** Please ensure serial monitor is set to transmit <newlines>

Message Logs:     Level 0
Status LED:       Pin 13
Device Control:   Pin 21
Sketch Version:   n/a
HomeSpan Version: 1.2.1
ESP-IDF Version:  v3.3.5-1-g85c43024c
ESP32 Board:      esp32
Sketch Compiled:  Jun  1 2021 05:38:29

Device Name:      HomeSpan LightBulb


*** HomeSpan Info ***

➤ Accessory:  AID=1
   ➟ Service AccessoryInformation:  IID=1, UUID=0x3E
      ⇨ Characteristic Name("My Table Lamp"):  IID=2, UUID=0x23
      ⇨ Characteristic Manufacturer("HomeSpan"):  IID=3, UUID=0x20
      ⇨ Characteristic SerialNumber("123-ABC"):  IID=4, UUID=0x30
      ⇨ Characteristic Model("120-Volt Lamp"):  IID=5, UUID=0x21
      ⇨ Characteristic FirmwareRevision("0.9"):  IID=6, UUID=0x52
      ⇨ Characteristic Identify(0):  IID=7, UUID=0x14
   ➟ Service HAPProtocolInformation:  IID=8, UUID=0xA2
      ⇨ Characteristic Version("1.1.0"):  IID=9, UUID=0x37
   ➟ Service LightBulb:  IID=10, UUID=0x43
      ⇨ Characteristic On(0):  IID=11, UUID=0x25

Configured as Bridge: NO

Service                         UUID         AID  IID  Update  Loop  Button  Linked Services
------------------------------  ----  ----------  ---  ------  ----  ------  ---------------
AccessoryInformation              3E           1    1      NO    NO      NO  -
HAPProtocolInformation            A2           1    8      NO    NO      NO  -
LightBulb                         43           1   10      NO    NO      NO  -

*** End Info ***

Generating SRP verification data for default Setup Code: 466-37-726
Setup Payload for Optional QR Code: X-HM://0052SD8YM    

Generating new random Accessory ID and Long-Term Ed25519 Signature Keys...
Initializing storage for Paired Controllers data...

Accessory ID:      EE:A7:C1:C4:4D:9D                               LTPK: F9F4D18516862B38BA0CE01EB748FABB21C604C3DC4E2B82755749F8A8EA8574
No Paired Controllers
Resetting Accessory Configuration number...

Accessory configuration has changed.  Updating configuration number to 1

*** WIFI CREDENTIALS DATA NOT FOUND.  YOU MAY CONFIGURE BY TYPING 'W <RETURN>'.

HomeSpan LightBulb is READY!





***@***.***
 
From: HomeSpan
Date: 2021-05-31 23:28
To: HomeSpan/HomeSpan
CC: Adajaken; Author
Subject: Re: [HomeSpan/HomeSpan] WiFi hotspot homespan doesn't appear (#84)
Can you check the following: From within the Arduino IDE Serial Monitor, type 'A ' when connected to HomeSpan. This should start the hot-spot and you should see the network HomeSpan-Setup show up under Settings->WiFi on your iPhone. If this does not work, can you verify that the ESP32 can connect to you WiFi network manually by using the 'W' command?
Also, can you send me the log file that appears in the Arduino IDE Serial Monitor when you first start the ESP32 (I assume you are using an ESP32, and not one of the newer ESP32S2 or ESP32C3 chips?

**Comment by HomeSpan** [comment]
Just to confirm - can you check you are following the instructions in the [User Guide](https://github.com/HomeSpan/HomeSpan/blob/master/docs/UserGuide.md#device-configuration-mode) exactly.   After start-up you should press and hold the Control Button (default is to attach to Pin 21) for 3 seconds.  Once the Status LED starts to rapidly flash, let go of the button.  This puts HomeSpan into Command Mode.  A message will be printed to the Arduino IDE confirming the device entered Command Mode, and the Status LED will blink, indicating Menu Choice 1 is selected.  Short-press the Control Button once to advance to Menu Choice 2 (the Status LED will start to slowly double-blink).  Then short-press the Control button a second time to advance to Menu Choice 3.  The Status LED should now slowly triple-blink, indicating Menu Choice 3 is selected, which is the Access Point.  Now you must Execute the Menu Command by pressing the Control Button for another 3 seconds and holding until the Status LED rapidly flashes.  Once you let go the Status LED should start to rapidly double-blink, indicating the Hotspot Access Point has been started.  A message on the Arduino IDE should confirm this, and after a few seconds it should also print out all the results of a network scan.

Note that once you enter Command Mode, you have 120 seconds to Execute a command before the Command mode exits automatically.  Also, once you Active the Hotspot Access Point you have 300 seconds to configure before it automatically terminates.

**Comment by Adajaken** [comment]
I didn't hold the control button for 3s. Now problem is solved. Thanks very much for your support.

Other question is, when I want to control an actual switch such as relay, which pin should I connect to? And where in program to define it? 

BR,



***@***.***
 
From: HomeSpan
Date: 2021-06-01 19:12
To: HomeSpan/HomeSpan
CC: Adajaken; Author
Subject: Re: [HomeSpan/HomeSpan] WiFi hotspot homespan doesn't appear (#84)
Just to confirm - can you check you are following the instructions in the User Guide exactly. After start-up you should press and hold the Control Button (default is to attach to Pin 21) for 3 seconds. Once the Status LED starts to rapidly flash, let go of the button. This puts HomeSpan into Command Mode. A message will be printed to the Arduino IDE confirming the device entered Command Mode, and the Status LED will blink, indicating Menu Choice 1 is selected. Short-press the Control Button once to advance to Menu Choice 2 (the Status LED will start to slowly double-blink). Then short-press the Control button a second time to advance to Menu Choice 3. The Status LED should now slowly triple-blink, indicating Menu Choice 3 is selected, which is the Access Point. Now you must Execute the Menu Command by pressing the Control Button for another 3 seconds and holding until the Status LED rapidly flashes. Once you let go the Status LED should start to rapidly double-blink, indicating the Hotspot Access Point has been started. A message on the Arduino IDE should confirm this, and after a few seconds it should also print out all the results of a network scan.
Note that once you enter Command Mode, you have 120 seconds to Execute a command before the Command mode exits automatically. Also, once you Active the Hotspot Access Point you have 300 seconds to configure before it automatically terminates.

**Comment by HomeSpan** [comment]
You may use any free pin to connect devices, such as relays, switches, etc.  The tutorials take you through all the steps starting in Example 5 (a Working LED).

**Comment by Adajaken** [comment]
Dear,

Another question needs your kindly help.
In Example WorkingLED, I added below codes and used Pin 33, however, when I try to control, the effect is not right, it's totally opposite. When I turn on lights in iPhone, the light actually turns of. and when turn off in iPhone, the real light is on. How to solve?
One more thing, Pin 16 and 17's funciton is normal. 





***@***.***
 
From: HomeSpan
Date: 2021-06-02 09:46
To: HomeSpan/HomeSpan
CC: Adajaken; Author
Subject: Re: [HomeSpan/HomeSpan] WiFi hotspot homespan doesn't appear (#84)
You may use any free pin to connect devices, such as relays, switches, etc. The tutorials take you through all the steps starting in Example 5 (a Working LED).

**Comment by Adajaken** [comment]
Also, after running WorkingLED for a while, the control will become very slow, no matter in LAn or in Internet, meanwhile, other HomeKit devices are working normal, no such slow problem. Why is that? (Control delay could reach 7 seconds sometime) 



***@***.***
 
From: ***@***.***
Date: 2021-06-03 09:56
To: HomeSpan/HomeSpan; HomeSpan/HomeSpan
CC: Author
Subject: Re: Re: [HomeSpan/HomeSpan] WiFi hotspot homespan doesn't appear (#84)
Dear,

Another question needs your kindly help.
In Example WorkingLED, I added below codes and used Pin 33, however, when I try to control, the effect is not right, it's totally opposite. When I turn on lights in iPhone, the light actually turns of. and when turn off in iPhone, the real light is on. How to solve?
One more thing, Pin 16 and 17's funciton is normal. 





***@***.***
 
From: HomeSpan
Date: 2021-06-02 09:46
To: HomeSpan/HomeSpan
CC: Adajaken; Author
Subject: Re: [HomeSpan/HomeSpan] WiFi hotspot homespan doesn't appear (#84)
You may use any free pin to connect devices, such as relays, switches, etc. The tutorials take you through all the steps starting in Example 5 (a Working LED).

**Reply by HomeSpan** [reply]
Hmm - there should be no differences between pins 33 and 16 or 17.  Can you describe exactly what you are connecting (a relay?).

As for the delays, can you try running just the very first Example 1 without modification?  If after running this for a while you start to get delays, we know the issue is related to connectivity.  If instead Example 1 runs without delays, the issue is related to the specifics of your code, which makes it easier to debug.
